src/main/java/com/uid2/admin/vertx/service/SiteService.java

```diff
@@ -296,11 +304,48 @@ private void handleSiteDomains(RoutingContext rc) {
         }
     }
 
+    private void handleSiteUpdate(RoutingContext rc) {
+        try {
+            // refresh manually
+            siteProvider.loadContent();
+
+            final Site existingSite = RequestUtil.getSite(rc, "id", siteProvider);
+            if (existingSite == null) {
+                return;
+            }
+            String description = rc.queryParam("description").stream().findFirst().orElse(null);
+            String visibleParam = rc.queryParam("visible").stream().findFirst().orElse(null);
+
+            if (description != null) {
+                existingSite.setDescription(description);
+            }
+            if (visibleParam != null) {
+                if ("true".equalsIgnoreCase(visibleParam)) {
+                    existingSite.setVisible(true);
+                } else if ("false".equalsIgnoreCase(visibleParam)) {
+                    existingSite.setVisible(false);
+                } else {
+                    ResponseUtil.error(rc, 400, "Invalid parameter for visible: " + visibleParam);
```

Comment:
I saw in your video when you `/list/sites`, some sites have `visible` set to null, I guess it's because we don't have that value in the json files in integ/prod environment. Should we update the json files for them? Otherwise would it throw a 400?

Reply:
> I saw in your video when you /list/sites, some sites have visible set to null, I guess it's because we don't have that value in the json files in integ/prod environment.
yeah - to be clear, this was tested on local, so it's because the sites.json for localstack does not have a `visible` value. Indeed, all values in integ/prod environments will have a null `visible` field.

> Should we update the json files for them? Otherwise would it throw a 400?
I don't think this is strictly necessary today - the `visible` parameter is not technically used anywhere yet. Updating a site using this method will not throw a 400 if `visible` is null, since we check it above.

That all being said, it probably makes sense to update this value for existing sites integ/prod so it is accurate, but it should not break anything if we don't. 

What are your thoughts?

Reply:
Cool, if it doesn't break anything in integ and production with the null visible value I'm happy to see it as a separate task :D Can you create a ticket for that please?